### PR TITLE
Training/2021.10/065 type derivation

### PR DESCRIPTION
**Pull request by leogermond**
Refacto for preparing 2021.10 EU OPAT

### CODE REVIEW

File: courses/fundamentals_of_ada/065_type_derivation.rst [hunk 1]
```diff
@@ -183,37 +183,18 @@ Simple Type Derivation
...
-           is null;
-        V1 : Parent;
-        V2 : Child;
-     begin
-        Not_A_Primitive (V1);
-        Not_A_Primitive (Parent (V2));
-     end Main2;
+     type Parent is range 1 .. 10;
+     procedure Prim (V : Parent);
+     type Child is new Parent;  -- Freeze
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/065_type_derivation.rst:192]: "Freeze Parent"

File: courses/fundamentals_of_ada/065_type_derivation.rst [hunk 2]
```diff
@@ -336,17 +321,17 @@ Implicit Subtype
 
    .. code:: Ada
 
-      type <Anon> is new Predefined_Integer_Type;
-      subtype T is <Anon> range L .. R;
+      type T is new Predefined_Integer_Type;
+      subtype T is T range L .. R;
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/065_type_derivation.rst:325]: Same name for type and subtype (call the first one "T_Prime?")

File: courses/fundamentals_of_ada/065_type_derivation.rst [hunk 3]
```diff
@@ -355,19 +340,19 @@ Implicit Subtype Explanation
 
 * New anonymous type `Anon` is derived from the predefined type
 * `Anon` inherits the type's operations (``+``, ``-`` ...)
-* `T`, subtype of `Anon` is created with range L .. R
-* :ada:`T'Base` will return the type `Anon`
+* `Sub`, subtype of `Anon` is created with range L .. R
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/065_type_derivation.rst:343]: You are now referencing "Anon" which you no longer have

File: courses/fundamentals_of_ada/065_type_derivation.rst (hunk 2 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/065_type_derivation.rst]: Done

**Review by frank-at-adacore — DISMISSED** [review]

File: courses/fundamentals_of_ada/065_type_derivation.rst (hunk 1 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/065_type_derivation.rst]: Done

File: courses/fundamentals_of_ada/065_type_derivation.rst [hunk 4]
```diff
@@ -330,44 +315,44 @@ Implicit Subtype
...
 --------------------------
 Integer Types Base Warning
 --------------------------
 
 .. code:: Ada
 
-   type <Anon> is new Predefined-Integer-Type;
-   subtype T is <Anon> range L .. R;
+   type Typ is new Predefined-Integer-Type;
+   subtype Sub is Typ range L .. R;
```
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/065_type_derivation.rst]: Done

File: courses/fundamentals_of_ada/065_type_derivation.rst [hunk 5]
```diff
@@ -330,44 +315,44 @@ Implicit Subtype
...
 ----------------------------
 Implicit Subtype Explanation
 ----------------------------
 
 .. code:: Ada
 
-   type <Anon> is new Predefined-Integer-Type;
-   subtype T is <Anon> range L .. R;
+   type Typ is new Predefined-Integer-Type;
+   subtype Sub is Typ range L .. R;
```
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/065_type_derivation.rst]: Done

File: courses/fundamentals_of_ada/065_type_derivation.rst [hunk 6]
```diff
@@ -330,44 +315,44 @@ Implicit Subtype
...
+      type Typ is range L .. R;
 
 * Is short-hand for
 
    .. code:: Ada
 
-      type <Anon> is new Predefined_Integer_Type;
-      subtype T is <Anon> range L .. R;
+      type Typ is new Predefined_Integer_Type;
+      subtype Sub is Typ range L .. R;
```
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/065_type_derivation.rst]: Done

**Review by frank-at-adacore — APPROVED** [review]

**Pull request merged**